pyre/extra_tests/parity_tests/raise_from_invalid_cause.py

```
@@ -0,0 +1,114 @@
+def a_cause_constructor_error_propagates():
+    """`raise Raised from CauseThatRaises` is the constructor's ValueError.
+
+    Whether `Raised()` has run by then is not asserted: `pyopcode.py:757-760`
+    instantiates the cause before popping the value, while the reference does
+    it the other way round, so the two disagree on that and only that.
+    """
+    try:
+        raise Raised from CauseThatRaises
+    except ValueError as error:
+        assert str(error) == "cause constructor", error
+    except BaseException as other:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `BaseException`

(BLE001)